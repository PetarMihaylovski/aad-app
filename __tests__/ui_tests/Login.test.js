import React from "react";
import {render, fireEvent, act} from '@testing-library/react-native';
import LoginScreen from "../../src/screens/auth/Login";


it('renders login screen', function () {
    const {getByText, getByTestId} = render(
        <LoginScreen/>
    );
    getByText("Email");
    getByText("Password");
    getByTestId("Login.login");
    getByTestId("Login.register");
});

it('logs-in successfully', async () => {
    fetch.mockResponseOnce('ok');
    const {getByTestId} = render(
        <LoginScreen/>
    );
    fireEvent.changeText(getByTestId("Login.emailInput"), "example@email.com");
    fireEvent.changeText(getByTestId("Login.passwordInput"), "asdgasdas");
    fireEvent.press(getByTestId("Login.login"));

    expect(fetch.mock.calls).toMatchSnapshot();
});

it('shows error state',  () => {
    const {getByTestId, queryAllByText} = render(
        <LoginScreen/>
    );
    fireEvent.changeText(getByTestId("Login.emailInput"), "example@email.com");
    fireEvent.changeText(getByTestId("Login.passwordInput"), "short");
    fireEvent.press(getByTestId("Login.login"));

    expect(queryAllByText('Unprocessable data entered!').length).toBe(1);
});