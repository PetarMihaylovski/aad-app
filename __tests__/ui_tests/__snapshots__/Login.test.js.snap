// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs-in successfully 1`] = `Array []`;
